### DISCUSSION TITLE
High-Level Goals for Publication

### DISCUSSION THREAD

**Discussion by mneinast** [Princeton-LSI-ResearchComputing/tracebase#1293] [General]
What should a published manuscript for TraceBase / version of TraceBase include?

# required
 - [x] upload peak annotation data
 - [x] download collated results in uniform formats
 - [x] upload mzXML
 - [x] download mzXML
 - [ ] a new public instance of tracebase, accessible to anyone, hosted at princeton
 - [ ] anyone can install tracebase and run their own server
 - [ ] updated documentation
 - [ ] polish of existing features
 
# considering
- [ ] support automated data
  - this could be a simple "workaround" support, based on identifying the type of PeakAnnotation file, or it could be something more in-depth
- [ ] shiny app for browsing data downloaded from tracebase (ie https://github.com/MRMacArthur/TracebaseView)

# save for after publication, if ever
 - [ ] "unknown" peaks

**Comment by hepcat72** [comment]
I'll propose some rough draft item breakdowns... and I'll think about these as I handle the milestone editing.

- download mzXML
  - the bulk download PR is currently in the review process and working on dev (I think it's on the branch that's on there now)
  - There are a number of extant issues for both display and download of mzXML files in other views
- a new public instance of tracebase, accessible to anyone, hosted at princeton
  - Probably fixing of data issues on the rabinowitz repo would fall here.  I'll create a publication milestone on that repo
- anyone can install tracebase and run their own server
  - Contrib doc needs updating
  - We should add a note, perhaps in the footer, about citing tracebase.  I forget the name of those software citations you can create.  We did it for TreeView.  I'll have to look in my notes.
- updated documentation
  - Code documentation
    - The handling of errors by the `AggregatedErrors` class needs more documentation
    - `MaintainedModel` needs more documentation
    - The advanced search "formats" need more documentation
  - Usage documentation
    - This is the "current" TraceBaseDocs
  - Curator documentation
    - We essentially need a generalized version of what's in nplcadmindocs
- polish of existing features
  - Bug fixes
  - Aesthetic display issues (typos, consistent headers, etc)
  - Slowness / timeouts
  - Optimization (loading speed)
  - Tests (convert tests from legacy code to current code)
